### ISSUE TITLE
Walrus Sites Service Worker might be causing duplicate POST requests going to our API backend

### ISSUE THREAD

**Issue by kkomelin** [MystenLabs/walrus-sites#217]
### What happened?

We noticed that the Walrus Sites service worker script might be causing receiving two duplicate POST requests by our API backend.

Please see the screenshots below:

**Local (without Walrus Sites Service Worker):**
![duplicate-request-sw-issue-dev](https://github.com/user-attachments/assets/1fcbf243-cebc-4a91-85f3-7f9485d1e5c3)

**[Live](https://icearrow.walrus.site/) (with Walrus Sites Service Worker):**
![duplicate-request-sw-issue-live](https://github.com/user-attachments/assets/c202478b-10fe-4b31-8b57-0c4fc5647e82)

### Search terms

please not now )

### Steps to reproduce

1. Open our app https://icearrow.walrus.site/ in Chrome
2. Open Chrome Dev Tools -> Network -> Fetch/XHR tab
3. Enter a string of text into the input field on the app frontpage, for example "test"
4. Press "Encrypt Secret" button
5. Find two requests to the backend endpoint /secrets (one goes directly to the backend and another goes there through the service worker).

### Deviation steps

We setup the latest Walrus through Docker:
https://github.com/b3b/icearrow-server/blob/main/walrus/Dockerfile

Our backend: https://github.com/b3b/icearrow-server
Our fontend: https://github.com/kkomelin/icearrow

### Relevant log output

```shell
From our Prometeus panel:

e","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726438664.6228857,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726438654.2551086,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726438664.62449,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726438654.2548952,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726438761.048407,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726438750.0163386,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726438762.0905778,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726438750.0163245,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726473329.6528993,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726473319.2488277,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726473329.6541584,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726473319.2487848,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726473607.4353135,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726473597.8830597,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726473607.437381,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726473597.8825104,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726474608.41964,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726474598.3832538,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726474608.4214537,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726474598.383292,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726489737.4007201,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726489725.7981348,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":200,"responseSize":50}
ice-hippos-1  | {"level":"info","ts":1726489753.4238338,"caller":"server/logging.go:42","msg":"Request handled","host":"172.21.0.1","timestamp":1726489725.7981267,"method":"POST","uri":"/secret","protocol":"HTTP/1.0","responseStatus":500,"responseSize":50}
```
```


### Version

- [X] `walrus` and `site-builder` are up to date.


### If applicable, what browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by giac-mysten** [comment]
Very odd, and interesting, thanks for the report @kkomelin . Looking into this.

**Comment by giac-mysten** [comment]
Quick question for debugging @kkomelin :
I see the "duplicate" on chrome, but I cannot find it on firefox or safari. Do you see duplicates in the logs even when using those browsers?
Could it be that chrome just logs the same request twice? (see this [SO question for example](https://stackoverflow.com/questions/61186729/what-to-change-to-prevent-double-request-from-service-worker)).

**Comment by kkomelin** [comment]
@giac-mysten I have tested in Chrome and Firefox and our server panel shows the same 2 request increase after each test regardless the browser. You may monitor the requests [here](https://api.icearrow.xyz/prometheus-ae306659/graph?g0.expr=yopass_http_requests_total%7Bmethod%3D%22POST%22%2C%20path%3D~%22%2F(secret)%22%2C%20code%3D%22200%22%7D&g0.tab=1&g0.display_mode=lines&g0.show_exemplars=0&g0.range_input=6h)

**Comment by giac-mysten** [comment]
@kkomelin you're right, found the culprit and solved in #218 .
Many thanks @Jordan-Mysten for the help and intuition!

**Comment by kkomelin** [comment]
Awesome job guys @giac-mysten @Jordan-Mysten! Thank you very much for your efforts!
